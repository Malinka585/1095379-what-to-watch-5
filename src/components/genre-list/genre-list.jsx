import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {incrementGenre} from "../../store/action";
import {getGenre, getGenreList} from "../../store/selectors";

const GenreList = (props) => {
  const {genreList, genre, changeGenre} = props;

  return (
    <ul className="catalog__genres-list">
      {genreList.map((genreItem, i) => (
        <li key={`${i}`}
          className={`catalog__genres-item ${genreItem === genre ? `catalog__genres-item--active` : ``} `}>
          <a
            href="#"
            className="catalog__genres-link"
            onClick={(evt) => {
              evt.preventDefault();
              changeGenre(genreItem);
            }}
          >
            {genreItem}</a>
        </li>
      ))}
    </ul>
  );
};

GenreList.propTypes = {
  genreList: PropTypes.array.isRequired,
  genre: PropTypes.string.isRequired,
  changeGenre: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  genre: getGenre(state),
  genreList: getGenreList(state),
});

const mapDispatchToProps = (dispatch) => ({
  changeGenre(genre) {
    dispatch(incrementGenre(genre));
  },
});

export {GenreList};
export default connect(mapStateToProps, mapDispatchToProps)(GenreList);
