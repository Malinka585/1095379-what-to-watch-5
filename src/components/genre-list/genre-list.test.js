import React from "react";
import renderer from "react-test-renderer";
import {GenreList} from "./genre-list";

const genreList = [
  `All genres`,
  `Comedy`
];

const noop = () => {};


describe(`Should GenreList render correctly`, () => {
  it(`With All genres`, () => {
    const tree = renderer
      .create(
          <GenreList
            genreList={genreList}
            genre={`All genres`}
            onClick={noop}
            changeGenre={noop}
          />

      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With Comedy`, () => {
    const tree = renderer
      .create(
          <GenreList
            genreList={genreList}
            genre={`Comedy`}
            onClick={noop}
            changeGenre={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
