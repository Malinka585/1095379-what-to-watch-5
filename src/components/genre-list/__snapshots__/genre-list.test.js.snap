// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenreList render correctly With All genres 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item  "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
</ul>
`;

exports[`Should GenreList render correctly With Comedy 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item  "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
</ul>
`;
